### PR TITLE
fix: LEAP-192: Check avatar file extension

### PR DESCRIPTION
**Pull request by triklozoid**
### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: label_studio/users/urls.py [hunk 1]
```diff
@@ -19,14 +19,18 @@
...
-        serve,
-        kwargs={'document_root': join(settings.MEDIA_ROOT, settings.AVATAR_PATH)},
-    ),
     # Token
     path('api/current-user/reset-token/', api.UserResetTokenAPI.as_view(), name='current-user-reset-token'),
     path('api/current-user/token', api.UserGetTokenAPI.as_view(), name='current-user-token'),
     path('api/current-user/whoami', api.UserWhoAmIAPI.as_view(), name='current-user-whoami'),
 ]
+
+if not settings.CLOUD_FILE_STORAGE_ENABLED:
```
Comment (jombooth) [inline label_studio/users/urls.py:30]: Is the thinking here that production deployments always enable this setting, and therefore we can use `serve` in deployments that don't have cloud storage set up? I would suggest adding a comment explaining the rationale for gating this path behind this particular setting.

File: label_studio/users/functions.py [hunk 2]
```diff
@@ -33,9 +34,16 @@ def check_avatar(files):
...
+    # check file extension
+    ext = os.path.splitext(filename)[1].lstrip('.').lower()
+    if ext not in valid_extensions:
+        raise forms.ValidationError('Please upload a valid image file with extensions: JPEG, JPG, GIF, or PNG.')
+
     # validate content type
     main, sub = avatar.content_type.split('/')
-    if not (main == 'image' and sub.lower() in ['jpeg', 'jpg', 'gif', 'png']):
+    if not (main == 'image' and sub.lower() in valid_extensions):
         raise forms.ValidationError('Please use a JPEG, GIF or PNG image.')
```
Comment (jombooth) [inline label_studio/users/functions.py]: Optional: these parts could be separated from the rest of avatar validation functionality, for other image file validation usecases.

**Review by jombooth — APPROVED** [review]
Approval modulo adding comment explaining gating the avatar path on CLOUD_FILE_STORAGE_ENABLED

File: label_studio/users/urls.py (hunk 1 above)
Comment (triklozoid) [reply label_studio/users/urls.py:30]: When this setting is enabled, we upload avatars to the cloud storage and their url looks different, so we don't need `serve` and this path in that case.

**Comment by MihajloHoma** [comment]
/git merge develop
> Successfully pushed new changes:
> Merge remote-tracking branch 'origin/develop' into fb-LEAP-192 (37baffeb98a39ea16e83c4a5bc1d88af1efc8bdc)
>
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/6337012634)

**Review by MihajloHoma — APPROVED** [review]

**Pull request merged**